frontend/src/static/js/api/client.ts

```diff
@@ -113,6 +140,82 @@ export class APIClient {
     });
   }
 
+  // Internal client detail for constructing a FeatureResultOffsetCursor pagination token.
+  // Typically, users of the /v1/features endpoint should use the provided pagination token.
+  // However, this token can be used to facilitate a UI with where we have selectable page numbers.
+  // Disclaimer: External users should be aware that the format of this token is subject to change and should not be
+  // treated as a stable interface. Instead, external users should rely on the returned pagination token long term.
+  private createOffsetPaginationTokenForGetFeatures(offset: number): string {
+    return base64urlEncode(JSON.stringify({offset: offset}));
+  }
+
+  /**
+   * Returns one page of data.
+   * TODO: needs to be generalized further to
+   *   handle more variations of queries. */
+  public async getPageOfData<
+    ResponseType extends {
+      data: unknown[];
+      metadata?: {next_page_token?: string};
+    },
+    ParamsType extends Record<string, unknown>,
+  >(
+    path: PathsWithMethod<paths, 'get'>,
+    params: ParamsType = {} as ParamsType,
+    offset?: number,
+    pageSize?: number
+  ): Promise<ResponseType> {
+    const qsParams: ParamsType & PageParams = {
+      ...params,
+      page_token: offset
+        ? this.createOffsetPaginationTokenForGetFeatures(offset)
+        : undefined,
+      page_size: pageSize,
+    };
+
+    const {data, error} = await this.client.GET(path, {
+      ...temporaryFetchOptions,
+      params: {
+        query: qsParams,
+      },
+    });
+
+    if (error !== undefined) {
+      throw createAPIError(error);
+    }
+
+    return data as ResponseType;
+  }
```

Comment:
I know getPageOfData is meant to be flexible for more calls in the future. But one thing that we can do to help is pass in the pageToken.

```suggestion
  public async getPageOfData<
    ResponseType extends {
      data: unknown[];
      metadata?: {next_page_token?: string};
    },
    ParamsType extends Record<string, unknown>,
  >(
    path: PathsWithMethod<paths, 'get'>,
    params: ParamsType = {} as ParamsType,
    pageToken?: string,
    pageSize?: number
  ): Promise<ResponseType> {
    const qsParams: ParamsType & PageParams = {
      ...params,
      page_token: pageToken,
      page_size: pageSize,
    };

    const {data, error} = await this.client.GET(path, {
      ...temporaryFetchOptions,
      params: {
        query: qsParams,
      },
    });

    if (error !== undefined) {
      throw createAPIError(error);
    }

    return data as ResponseType;
  }
```

Reply:
I agree with trying to avoid type assertions, but the "as ParamsType" and "as ResponseType" assertions seem to be required still, according to vscode anyway.  I get red squigly lines under the {} and return otherwise.

Reply:
Regarding the page_token, I currently rely on the numeric offset instead, Do you mean that if there is a pageToken parameter to getPageOfData, then we use that instead of the offset parameter?  I'd rather get this much submitted and figure out how to migrate to use the pageToken in a subsequent PR.

Reply:
Reviewing this, we will need a different getAllPages that returns an AsyncIterator. Maybe call it getAllPagesIterator. That way getFeatureCountsForBrowser and getFeatureStatsByBrowserAndChannel can use it because they cannot use getAllPagesOfData

I pushed a commit to reduce the number of type assertions to only one. We can address the getAllPagesIterator later.